frontend/src/actions/budget.ts

```diff
@@ -0,0 +1,88 @@
+import { Dispatch } from 'redux';
+import { AppState } from 'src/types';
+import axios, { AxiosResponse, AxiosError } from 'axios';
+import { loadLocalStorageItem } from 'src/helpers/localStorage';
+import { APP_STORAGE_KEY, REQUEST, FAILURE, SUCCESS } from 'src/constants';
+import {
+  FETCH_TRANSACTIONS,
+  FETCH_CATEGORIES,
+  SAVE_TRANSACTIONS,
+} from 'src/constants/budget';
+import { TransactionRequest } from 'src/types/budget';
+
+export const init = () => {
+  // tslint:disable-next-line
+  return async (dispatch: Dispatch<any>) => {
+    await dispatch(fetchTransactions());
+    await dispatch(fetchCategories());
+  };
+};
+
+export const fetchTransactions = () => {
+  return (dispatch: Dispatch, getState: () => AppState) => {
+    dispatch({ type: FETCH_TRANSACTIONS[REQUEST] });
+    const { accessToken, userId } = loadLocalStorageItem(APP_STORAGE_KEY);
```

Comment:
If these turn out to be undefined or null, do we still need to make a call?
Would we automatically go back to login page?

Reply:
You wouldn't be able to make this call if you weren't authenticated because this view requires user to be signed in.